docs/golden-retriever.mdx

```diff
@@ -0,0 +1,139 @@
+  `Service Worker` can keep references to large files. Service Worker storage is
+  _quite_ temporary though, and doesn’t persist across browser crashes or
+  restarts. It works well, however, for accidental refreshes or closed tabs.
+
+Upon restore, the plugin first restores state from `LocalStorage` and then
+checks both file storages — `IndexedDB` and `ServiceWorker` — merging the
+results.
+
+If restore is unsuccessful for certain files, they will be marked as “ghosts” in
+the Dashboard UI, and a message + button offering to re-select those files will
+be displayed.
+
+## When should I use this?
+
+When you want extra insurance for the user-selected files. If you feel like
+users might spend some time picking files, tweaking descriptions etc, and will
+not appreciate having to do it over again if something crashes. Another use case
+might be when you think user might want to pick a few files, navigate away to do
+something else in your app or otherwise, and then come back to the same
+selection.
+
+## Install
+
+<Tabs>
+  <TabItem value="npm" label="NPM" default>
+
+```shell
+npm install @uppy/golden-retriever
+```
+
+  </TabItem>
+
+  <TabItem value="yarn" label="Yarn">
+
+```shell
+yarn add @uppy/golden-retriever
+```
+
+  </TabItem>
+
+  <TabItem value="cdn" label="CDN">
+    <UppyCdnExample>
+      {`
+        import { Uppy, GoldenRetriever } from "{{UPPY_JS_URL}}"
+        new Uppy().use(GoldenRetriever)
+      `}
+    </UppyCdnExample>
+  </TabItem>
+</Tabs>
+
+## Use
+
+```js {9} showLineNumbers
+import Uppy from '@uppy/core';
+import Dashboard from '@uppy/dashboard';
+import Tus from '@uppy/tus';
+import GoldenRetriever from '@uppy/golden-retriever';
+
+new Uppy().use(Dashboard).use(Tus).use(GoldenRetriever);
+```
+
+By default, Golden Retriever will only use the `IndexedDB` storage, which is
+good enough for files up to 5 megabytes (configurable). `Service Worker` is
+optional and requires setup.
+
+### Enabling Service Worker
+
+With the Service Worker storage, Golden Retriever will be able to temporary
+store references to large files.
+
+1\. Bundle your own service worker `sw.js` file with Uppy GoldenRetriever’s
+service worker.
+
+:::tip For Webpack, check out
+[serviceworker-webpack-plugin](https://github.com/oliviertassinari/serviceworker-webpack-plugin).
+:::
+
+```js
+// sw.js
+
+import('@uppy/golden-retriever/lib/ServiceWorker');
+```
+
+2\. Register it in your app’s entry point:
+
+```js
+// you app.js entry point
+uppy.use(GoldenRetriever, { serviceWorker: true });
+
+if ('serviceWorker' in navigator) {
+	navigator.serviceWorker
+		.register('/sw.js') // path to your bundled service worker with GoldenRetriever service worker
+		.then((registration) => {
+			console.log(
+				'ServiceWorker registration successful with scope: ',
+				registration.scope,
+			);
+		})
+		.catch((error) => {
+			console.log(`Registration failed with ${error}`);
+		});
+}
+```
```

Comment:
````suggestion
2. Register it in your app’s entry point:

   ```js
   // you app.js entry point
   uppy.use(GoldenRetriever, { serviceWorker: true });

   if ('serviceWorker' in navigator) {
   	navigator.serviceWorker
   		.register('/sw.js') // path to your bundled service worker with GoldenRetriever service worker
   		.then((registration) => {
   			console.log(
   				'ServiceWorker registration successful with scope: ',
   				registration.scope,
   			);
   		})
   		.catch((error) => {
   			console.log(`Registration failed with ${error}`);
   		});
   }
   ```
````